### PR TITLE
Use regex to update config. Add/update tests

### PR DESCRIPTION
**Pull request by twangboy**
Uses regex to search and replace the various minion configs

Existing config always remains untouched. Default config and Custom config are updated with values from the GUI/CLI.

Default values for minion config are:
- master: salt
- id: hostname
If those values are found, do not update the config file. This is inline with how the Nullsoft installer works.

Handle relative paths to the Custom Config

Write tests for the various config scenarios.

Resolves issue: https://github.com/saltstack/salt-windows-msi/issues/115

### CODE REVIEW

File: CustomAction01/CustomAction01.cs [hunk 1]
```diff
@@ -148,7 +148,7 @@ public class MinionConfiguration : WixExtension {
                     session.Log("...MASTER set to salt because it was unset and CONFIG_TYPE=Default");
                 }
                 if (session["MINION_ID"] == "") {
-                    session["MINION_ID"] = Environment.MachineName;
+                    session["MINION_ID"] = "hostname";
```
Comment (marbx) [inline CustomAction01/CustomAction01.cs:151]: Wouldn't that result in the literal "hostname"?

File: CustomAction01/CustomAction01.cs [hunk 2]
```diff
@@ -172,7 +172,7 @@ public class MinionConfiguration : WixExtension {
                         session.Log("...MINION_ID set to kept config ");
                         session["MINION_ID"] = id_from_previous_installation;
                     } else {
-                        session["MINION_ID"] = Environment.MachineName;
+                        session["MINION_ID"] = "hostname";
```
Comment (marbx) [inline CustomAction01/CustomAction01.cs:175]: Wouldn't that result in the literal "hostname"?

Comment (marbx) [reply CustomAction01/CustomAction01.cs:175]: OK, got it

File: CustomAction01/CustomAction01.cs [hunk 3]
```diff
@@ -525,23 +542,43 @@ public class MinionConfiguration : WixExtension {
...
+                    session.Log("...CONFIG_TYPE: Custom, copying custom config");
+                    save_custom_config_file_DECAC(session);
+                    break;
+                case "Default":
+                    // This is just a placeholder for CONFIG_TYPE=Default
+                    session.Log("...CONFIG_TYPE: Default, using default config");
+                    break;
+                default:
+                    session.Log("...UNKNOWN CONFIG_TYPE: " + config_type);
+                    // Not sure if this is a valid ActionResult, but we need to die here
```
Comment (marbx) [inline CustomAction01/CustomAction01.cs:577]: The installation will fail

File: build.ps1 [hunk 4]
```diff
@@ -67,8 +67,8 @@ if (ProductcodeExists "{03368010-193D-4AE2-B275-DD2EB32CD427}") {
     VerifyOrDownload $wixInstaller `
         "https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311.exe" `
         "32BB76C478FCB356671D4AAF006AD81CA93EEA32C22A9401B168FC7471FECCD2"
-    Write-Host -ForegroundColor Yellow "    *** Please install the Wix toolset (by clicking the fancy Install button) ***"
-    Start-Process $wixInstaller -Wait -NoNewWindow
+    Write-Host -ForegroundColor Yellow "    *** Installing the Wix toolset ***"
+    Start-Process $wixInstaller -ArgumentList "/install","/quiet","/norestart" -Wait -NoNewWindow
```
Comment (marbx) [inline build.ps1:71]: Just saw this on a new dev machine, nice!

**Comment by marbx** [comment]
- Great use of (multiline) regex and many code improvements. 
- Test suites grown, very good.

File: build.ps1 [hunk 5]
```diff
@@ -55,12 +55,8 @@ if (ProductcodeExists "{03368010-193D-4AE2-B275-DD2EB32CD427}") {
     $dotnet3state = (Get-WindowsOptionalFeature -Online -FeatureName "NetFx3").State
     $dotnet3enabled = $dotnet3state -Eq "Enabled"
     if (-Not ($dotnet3enabled)) {
-        Write-Host -ForegroundColor Yellow "    ***  Please enable .Net Framework 3.5 (rquired for or WiX 3)***"
-        if([Environment]::OSVersion.Version -ge (new-object 'Version' 10,0)) {
-          # Windows 10 or higher
-          Start-Process optionalfeatures -Wait -NoNewWindow
-        }
-        Read-Host -Prompt "Press any key to continue"
+        Write-Host -ForegroundColor Yellow "    ***  Enabling Feature .Net Framework 3.5 (rquired for or WiX 3) ***"
```
Comment (marbx) [inline build.ps1:58]: typo rquired ... required

Comment (twangboy) [reply build.ps1:58]: Fixed

**Pull request merged**